src/pages/workspace/workflows/approvals/WorkspaceWorkflowsApprovalsExpensesFromPage.tsx

```diff
@@ -1,28 +1,169 @@
+            const availableMembers = approvalWorkflow.availableMembers
+                .map((member) => {
+                    const isAdmin = policy?.employeeList?.[member.email].role === CONST.REPORT.ROLE.ADMIN;
+                    const policyMemberEmailsToAccountIDs = PolicyUtils.getMemberAccountIDsForWorkspace(policy?.employeeList);
+                    const accountID = Number(policyMemberEmailsToAccountIDs[member.email] ?? '');
+
+                    return {
+                        text: member.displayName,
+                        alternateText: member.email,
+                        keyForList: member.email,
+                        isSelected: false,
+                        login: member.email,
+                        icons: [{source: member.avatar ?? FallbackAvatar, type: CONST.ICON_TYPE_AVATAR, name: member.displayName, id: accountID}],
+                        rightElement: isAdmin ? <Badge text={translate('common.admin')} /> : undefined,
+                    };
+                })
+                .filter((member) => !selectedMembers.some((selectedOption) => selectedOption.login === member.login));
+
+            members.push(...availableMembers);
+        }
+
+        const filteredMembers =
+            debouncedSearchTerm !== ''
+                ? members.filter((option) => {
+                      const searchValue = OptionsListUtils.getSearchValueForPhoneOrEmail(debouncedSearchTerm);
+                      const isPartOfSearchTerm = !!option.text?.toLowerCase().includes(searchValue) || !!option.login?.toLowerCase().includes(searchValue);
+                      return isPartOfSearchTerm;
+                  })
+                : members;
+
+        return [
+            {
+                title: undefined,
+                data: filteredMembers,
+                shouldShow: true,
+            },
+        ];
+    }, [approvalWorkflow?.availableMembers, debouncedSearchTerm, policy?.employeeList, selectedMembers, translate]);
+
+    const nextStep = useCallback(() => {
+        const members: Member[] = selectedMembers.map((member) => ({displayName: member.text, avatar: member.icons?.[0]?.source, email: member.login}));
+        Workflow.setApprovalWorkflowMembers(members);
+
+        const firstApprover = approvalWorkflow?.approvers?.[0]?.email;
+        if (approvalWorkflow?.isBeingEdited && firstApprover) {
+            Navigation.navigate(ROUTES.WORKSPACE_WORKFLOWS_APPROVALS_EDIT.getRoute(route.params.policyID, firstApprover));
+        } else {
+            Navigation.navigate(ROUTES.WORKSPACE_WORKFLOWS_APPROVALS_APPROVER.getRoute(route.params.policyID));
+        }
+    }, [approvalWorkflow?.approvers, approvalWorkflow?.isBeingEdited, route.params.policyID, selectedMembers]);
+
+    const goBack = useCallback(() => {
+        if (!approvalWorkflow?.isBeingEdited) {
+            Workflow.clearApprovalWorkflow();
+        }
+        Navigation.goBack();
+    }, [approvalWorkflow?.isBeingEdited]);
```

Comment:
Clearing the approval workflow on going back caused the not found view to appear briefly while hiding the modal (https://github.com/Expensify/App/issues/50094). Also this `goBack` function is not called if we dismiss the modal on outside-click.

We went with a different approach that is to clear the workflow (from `WorkspaceWorkflowsPage`) before navigating to this page.